salt/tests/unit/modules/test_metalk8s_kubernetes.py

```diff
@@ -0,0 +1,566 @@
+                            called_with=None, **kwargs):
+        """
+        Tests the return of `repace_object` function
+        """
+        def _replace_mock(body, **_):
+            if raise_replace:
+                raise ApiException(status=0, reason='An error has occurred')
+
+            # body == object
+            return body
+
+        get_kind_info_mock = MagicMock()
+        if not info_scope:
+            get_kind_info_mock.side_effect = ValueError(
+                'An error has occurred'
+            )
+        get_kind_info_mock.return_value.scope = info_scope
+
+        replace_mock = get_kind_info_mock.return_value.client.replace
+        replace_mock.side_effect = _replace_mock
+
+        manifest_read_mock = MagicMock()
+        # None = IOError
+        # False = YAMLError
+        if manifest_file_content is None:
+            manifest_read_mock.side_effect = IOError(
+                'An error has occurred'
+            )
+        elif manifest_file_content is False:
+            manifest_read_mock.side_effect = yaml.YAMLError(
+                'An error has occurred'
+            )
+        else:
+            manifest_read_mock.return_value = manifest_file_content
+
+        utils_dict = {
+            'metalk8s_kubernetes.get_kind_info': get_kind_info_mock
+        }
+        salt_dict = {
+            'metalk8s_kubernetes.read_and_render_yaml_file': manifest_read_mock
+        }
+        with patch.dict(metalk8s_kubernetes.__utils__, utils_dict), \
+                patch.dict(metalk8s_kubernetes.__salt__, salt_dict):
+            if raises:
+                self.assertRaisesRegexp(
+                    Exception,
+                    result,
+                    metalk8s_kubernetes.replace_object,
+                    **kwargs
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes.replace_object(**kwargs),
+                    result
+                )
+                replace_mock.assert_called_once()
+                if called_with:
+                    self.assertDictContainsSubset(
+                        called_with,
+                        replace_mock.call_args.kwargs
+                    )
+
+    @parameterized.expand(
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES['get_object'] + YAML_TESTS_CASES['common_tests']
+    )
+    def test_get_object(self, result, raises=False, raise_retrieve=False,
+                        info_scope="namespaced", manifest_file_content=None,
+                        called_with=None, **kwargs):
+        """
+        Tests the return of `get_object` function
+        """
+        def _retrieve_mock(name, **_):
+            if raise_retrieve:
+                if raise_retrieve is True:
+                    raise ApiException(
+                        status=0,
+                        reason='An error has occurred'
+                    )
+                else:
```

Comment:
detail (same for `_delete_mock` above):
```suggestion
            if raise_retrieve is True:
                raise ApiException(
                    status=0,
                    reason='An error has occurred'
                )
            elif raise_retrieve:
```
And since this variable seems to define the status code returned by the API, I would rather named it like `api_status_code` (or something else) and set it to `None` by default.
```suggestion
            if api_status_code is not None:
                raise ApiException(
                    status=api_status_code,
                    reason='An error has occurred'
                )
```
But you may want to keep it the way it's done in other test cases, just suggesting.

Reply:
You right, it's clearer